packages/react-components/react-tags/src/components/Tag/Tag.types.ts

```diff
@@ -1,17 +1,30 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { Avatar } from '@fluentui/react-avatar';
 
 export type TagSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'div'>>;
+  content?: Slot<'span'>;
+  avatar?: Slot<typeof Avatar>;
```

Comment:
yeah, this is a big question. If we did do it that way, it would need to be a TagAvatar because there are a few style changes that we'd want to apply to it.

If the component is never used, will the Avatar get shaken out?

Reply:
Yeah, I had hoped that Avatar could be treeshaken out if it wasn't used, but it seems that is not the case. We'll probably go the <TagAvatar /> route because there will be some style mods we'll need to make, and some props to ommit.

Reply:
Was thinking about this a little more and have two suggestions:

1. Let's compare Tag's bundle size with Avatar vs without. It should be smaller without but maybe the delta isn't too much to worry about (plus Avatar is a fairly common control so apps may already be using it)
2. What styles need to be applied to "TagAvatar"? Why not add these to Avatar itself?

Reply:
the layout of the presence badge needs to change a bit when the avatar is inside of a tag. Its minor, a few pixels, but necessary.

Reply:
Michah I just saw notice your message on group chat, that there're two issues with avatar, one is right margin, another is presence touching tag border. I guess if we sort out both issue with design, we're okay with using one slot?

Reply:
I found a neat solution for our avatar styling problem in #24790